client/src/components/Board/Board.tsx

```diff
@@ -0,0 +1,61 @@
+import { DragDropContext, Droppable, DropResult } from 'react-beautiful-dnd';
+import Column from '../Column/Column';
+import useStyles from './useStyles';
+import { CircularProgress, Grid } from '@material-ui/core';
+import { useBoardContext } from '../../context/useBoardContext';
+import { Column as ColumnInterface } from '../../interface/Board';
+import { moveColumn, moveCard } from '../../helpers/boardContext/boardContext';
+import ColumnAddButton from '../ColumnAddButton/ColumnAddButton';
+
+const Board = (): JSX.Element => {
+  const classes = useStyles();
+
+  const { boardData, updateBoardData } = useBoardContext();
+
+  if (!boardData) {
+    return <CircularProgress />;
+  }
+
+  const handleDragEnd = (result: DropResult) => {
+    const { destination, source, type } = result;
+    if (!destination) {
+      return;
+    }
+
+    if (destination.droppableId === source.droppableId && destination.index === source.index) {
+      return;
+    }
+
+    if (type === 'column') {
+      updateBoardData(moveColumn(boardData, source.index, destination.index));
+      return;
+    } else {
+      updateBoardData(
+        moveCard(boardData, source.index, destination.index, source.droppableId, destination.droppableId),
+      );
+    }
+  };
+
+  return (
+    <Grid container xs={12} className={classes.root}>
+      <ColumnAddButton location={'left'} />
+      <Grid item xs={10}>
+        <DragDropContext onDragEnd={handleDragEnd}>
+          <Droppable droppableId="column-list" direction="horizontal" type="column">
```

Comment:
Q: Did you write this code or is the base of the branch wrong?

Reply:
I wrote this code, I tried my best to adapt the existing drag and drop feature but ended up rewriting the whole thing. It was influenced by the drag and drop feature that was merged in.